### ISSUE TITLE
Cannot set all properties when using Command in XAML layout

### ISSUE THREAD

**Issue by wjk** [picoe/Eto#1059]
## Expected Behavior

When I set the Command of a MenuItem from XAML, all properties should be set on the MenuItem instance.

## Actual Behavior

Only the click handler and the enabled property are set on the MenuItem.

## Steps to Reproduce the Problem

First, create a command and place it into a public static property, like so:

```csharp
public static class Commands
{
    public static Command SayHelloCommand
    {
        get
        {
            Command cmd = new Command();
            cmd.MenuText = "Say Hello";
            cmd.Shortcut = Keys.Application | Keys.S;
            return cmd;
        }
    }
}
```

Then create some XAML like this:

```xml
<MenuBar>
  <ButtonMenuItem Text="File">
    <ButtonMenuItem Command="{x:Static app:Commands.SayHelloCommand}" />
  </ButtonMenuItem>
</MenuBar>
```

When you run the app you will notice that the ButtonMenuItem has no title, nor is its shortcut set. To set these you must use the constructor that takes a `Command` instance, which cannot be used directly in XAML.

I tried submitting a PR that sets all properties when `MenuItem.Command` is set, but ran into a stumbling block regarding the different subclasses of `Command`. If you can give me a pointer on how to best implement this, I can get started on fixing it.

## Specifications

  - Version: 2.4.0, as available on NuGet
  - Platform(s): All (this is a core problem, not platform-specific).

**Comment by cwensley** [comment]
Hm, yeah it's not set up do work like that, but it would be nice if it worked.

This might actually work instead though, as it should be able to convert that instance to a ButtonMenuItem, and would also support all the different types of Command:

```xml
<MenuBar>
  <ButtonMenuItem Text="File">
    <x:Static Member="app:Commands.SayHelloCommand" />
  </ButtonMenuItem>
</MenuBar>
```

**Comment by cwensley** [comment]
@wjk did you try my suggestion?

**Comment by wjk** [comment]
@cwensley Sorry I haven't gotten back to you. I've started a new job over the last few weeks, so I haven't had much free time lately. Furthermore, I have not done any work with Eto for some time before that, due to this and other issues. I'll try to get to trying this soon. Thanks!

**Comment by wjk** [comment]
Your suggestion worked! However, due to other issues with Eto, I have scrapped the project that I was using it with since I first reported this issue. (If you're wondering what the issues were, it was integration with Cocoa on macOS that would (a) not break Eto's cross-platform architecture too bad while (b) looking and working like Mac users would expect. It's very difficult.) Thank you very much nevertheless!
